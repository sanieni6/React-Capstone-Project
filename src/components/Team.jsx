import React from 'react';
import PropTypes from 'prop-types';

const Team = ({ team, stats, getLogo }) => (
  <li className="stand" key={team.id}>
    {getLogo(team)}
    <div className="second-container">
      <p className="name">{team.name}</p>
      <div className="stats">
        <p className="points">
          Points:
          {stats[2].value}
        </p>
        <p className="wins">
          Wins:
          {stats[6].value}
        </p>
        <p className="losses">
          Losses:
          {stats[1].value}
        </p>

      </div>

    </div>

  </li>
);

Team.propTypes = {
  team: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
  stats: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.object,
  ]).isRequired,
  getLogo: PropTypes.func.isRequired,
};

export default Team;
