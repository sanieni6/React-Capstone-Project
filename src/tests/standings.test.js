import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Team from '../components/Team';

describe('Team component', () => {
  const teamData = {
    id: 1,
    name: 'Barcelona',
    crestUrl: 'https://example.com/barcelona-crest.png',
  };
  const statsData = [
    { type: 'Total', value: 75 },
    { type: 'Loses', value: 10 },
    { type: 'Draws', value: 5 },
    { type: 'Points', value: 60 },
    { type: 'Goals For', value: 80 },
    { type: 'Goals Against', value: 20 },
    { type: 'Wins', value: 20 },
  ];

  test('renders team name and stats correctly', () => {
    const { container } = render(<Team team={teamData} stats={statsData} getLogo={() => null} />);
    expect(container).toMatchSnapshot();
  });

  test('renders team logo', () => {
    const { getByAltText } = render(<Team
      team={teamData}
      stats={statsData}
      getLogo={(team) => <img src={team.crestUrl} alt={team.name} />}
    />);
    expect(getByAltText('Barcelona')).toHaveAttribute('src', 'https://example.com/barcelona-crest.png');
  });

  test('renders team stats correctly', () => {
    const { getByText } = render(<Team team={teamData} stats={statsData} getLogo={() => null} />);
    expect(getByText('Points:5')).toBeInTheDocument();
    expect(getByText('Wins:20')).toBeInTheDocument();
    expect(getByText('Losses:10')).toBeInTheDocument();
  });
});
