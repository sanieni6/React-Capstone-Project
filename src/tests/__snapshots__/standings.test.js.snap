// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team component renders team name and stats correctly 1`] = `
<div>
  <li
    class="stand"
  >
    <div
      class="second-container"
    >
      <p
        class="name"
      >
        Barcelona
      </p>
      <div
        class="stats"
      >
        <p
          class="points"
        >
          Points:
          5
        </p>
        <p
          class="wins"
        >
          Wins:
          20
        </p>
        <p
          class="losses"
        >
          Losses:
          10
        </p>
      </div>
    </div>
  </li>
</div>
`;
